Source/Extensions/Blazorise.DataGrid/_DataGridRow.razor.cs

```diff
@@ -55,13 +58,31 @@ protected internal async Task HandleClick( BLMouseEventArgs eventArgs )
                 && ParentDataGrid.SelectedRow != null
                 && (object)Item == (object)ParentDataGrid.SelectedRow )
             {
+                await Selected.InvokeAsync( default );
+            }
+            else if ( ParentDataGrid.MultiSelect && ParentDataGrid.SelectedRows is object && ParentDataGrid.SelectedRows.Any( x => (object)x == (object)Item ) )
+            {
+                //If the user selects an already selected multiselect row, seems like it should be more transparent, to just de-select both normal and multi selection
+                //Remove this, if that is not the case
                 await Selected.InvokeAsync( default );
             }
             else
             {
                 await Selected.InvokeAsync( Item );
             }
-        }
+
+            if ( ParentDataGrid.MultiSelect )
+            {
+                if ( multiSelect is object )

```

Comment:
Why do you check for object here? `multiSelect is object`

Reply:
It's a null check. Same as multiselect != null, just easier to read.
multiselect should be null if checkbox column functionality isn't being used.

Reply:
Its not so common to see the null check like that.

I think better is to have it like ussual(`name != null`) or in new c# 9

```
if (name is null) { }

if (name is not null) { }
```

Reply:
I'll fix it by using the same convention as you've been already using. `name != null`
Sincerely, between 
`name != null`
&&
`if (name is not null) { }`
I still like the old way the best... they're both "different then" kind of assertions, either way.